Lib/glyphsLib/builder/sources.py

```diff
@@ -53,6 +57,21 @@ def _to_designspace_source(self, master, is_regular):
         source.filename = build_ufo_path(
             '', source.familyName, source.styleName)
 
+        # Make sure UFO filenames are unique, lest we overwrite masters that
+        # happen to have the same weight name.
+        n = "_"
+        while any(s != source and s.filename == source.filename
+                  for s in self._sources.values()):
+            source.filename = os.path.basename(
+                build_ufo_path('.', source.familyName, source.styleName + n))
```

Comment:
why the "." here? A few lines before you called `build_ufo_path("", ...)` with an empty directory string.
You don't need to compute the filename again, just add the suffix to the current one `source.filename += n`

Reply:
Well, I wanted to compute `Something-Bold_.ufo` instead of `Something-Bold.ufo_`. I vaguely remember something choking when the suffix isn't ".ufo".